File: src/main/java/dev/akif/cookbook/ingredient/IngredientEntity.java
package dev.akif.cookbook.ingredient;

import jakarta.persistence.*;
import lombok.*;

@AllArgsConstructor
@Entity(name = "ingredient")
@Getter
@NoArgsConstructor
@Setter
@Table(name = "ingredients")
@ToString
public class IngredientEntity {
    @Column(name = "id", nullable = false, unique = true)
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false, unique = true)
    private String name;

    @Column(name = "vegetarian", nullable = false)
    private Boolean vegetarian;

    public Ingredient toIngredient() {
        return new Ingredient(id, name, vegetarian);
    }
}

File: src/main/kotlin/dev/akif/cookbook/ingredient/IngredientEntity.kt
package dev.akif.cookbook.ingredient

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table

@Table(name = "ingredients")
class IngredientEntity(
    @Id
    var id: Long? = null,
    var name: String = "",
    var vegetarian: Boolean = false
) {
    fun toIngredient(): Ingredient =
        Ingredient(
            id = id ?: -1,
            name = name,
            vegetarian = vegetarian
        )
}
